File: src/main/java/org/foo/models/Payment.java
package org.foo.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.foo.enums.Months;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "payments")
public class Payment extends BaseEntity {
  private int year;
  private BigDecimal amount;
  private LocalDate paymentDate;
  private boolean isPaid;
  private String companyCc;
  private String description;
  @Enumerated(EnumType.STRING)
  private Months month;
  @ManyToOne
  @JoinColumn(name = "company_id")
  private Company company;

  // Getters and setters
}

File: lab1-frontend/src/app/common/payment.ts
import {Months} from "../enums/months";
import {CompanyDto} from "./company";

export class Payment {
  constructor(
    public id: number,
    public year: number,
    public month: Months,
    public paymentDate: Date, // Use JavaScript Date for LocalDate
    public amount: number, // Use number for BigDecimal
    public isPaid: boolean,
    public companyCC: string,
    public description: string,
    public companyDto: CompanyDto
  ) {}
}
